File: src/main/java/alfheim/common/potion/PotionDeathMark.java
package alfheim.common.potion;

import alfheim.AlfheimCore;
import alfheim.common.core.util.AlfheimConfig;
import alfheim.common.core.util.DamageSourceSpell;
import net.minecraft.entity.EntityLivingBase;

public class PotionDeathMark extends PotionAlfheim {

	public PotionDeathMark() {
		super(AlfheimConfig.potionIDDeathMark, "deathMark", true, 0x553355);
	}

	public boolean isReady(int time, int ampl) {
		return time == 1; 
	}

	public void performEffect(EntityLivingBase living, int ampl) {
		if (AlfheimCore.enableMMO) living.attackEntityFrom(DamageSourceSpell.mark, Integer.MAX_VALUE);
	}
}

File: src/main/java/alfheim/common/potion/PotionDeathMark.kt
package alfheim.common.potion

import alfheim.common.core.handler.AlfheimConfigHandler
import alfheim.common.core.util.DamageSourceSpell
import alfheim.common.spell.darkness.SpellDeathMark
import net.minecraft.entity.EntityLivingBase

class PotionDeathMark: PotionAlfheim(AlfheimConfigHandler.potionIDDeathMark, "deathMark", true, 0x553355) {
	
	override fun isReady(time: Int, ampl: Int) = time == 1
	
	override fun performEffect(living: EntityLivingBase, ampl: Int) {
		if (AlfheimConfigHandler.enableMMO) living.attackEntityFrom(DamageSourceSpell.mark, SpellDeathMark.damage)
	}
}
